File: codeforces/archive/1883E/main.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>
#include <set>
#include <algorithm>
#include <cmath>

typedef long long ll;

int main() {
    int t;
    std::cin >> t;

    while (t--) {
        int n;
        std::cin >> n;

        std::vector<double> a(n);

        for (int i = 0; i < n; i++) {
            int j;
            std::cin >> j;

            a[i] = std::log2(j);
        }
        
        ll count = 0;

        for (int i = 1; i < n; i++) {
            ll diff = std::max(0ll, (ll) std::ceil(a[i - 1] - a[i]));
            count += diff;
            a[i] += diff;
        }

        std::cout << count << std::endl;
    }

    return 0;
}

File: codeforces/archive/1883E/main.py
from math import log2, ceil

t = int(input())

# I really hope floating point errors aren't doing me in because that would be
# severely unfortunate

# It seems to be so; that is severely severe
for _ in range(t):
    n = int(input())
    a = [Decimal(i).ln() / Decimal(2).ln() for i in input().split(" ")]
    print(a)
    
    count = 0

    for i in range(1, n):
        diff = int(max(0, ceil(a[i-1] - a[i])))
        count += diff
        a[i] += diff

    print(count)
